src/include/uapi/ipc/topology.h

```diff
@@ -162,6 +163,18 @@ struct sof_ipc_comp_volume {
 	uint32_t initial_ramp;	/**< ramp space in ms */
 } __attribute__((packed));
 
+/* generic selector component */
+struct sof_ipc_comp_selector {
+	struct sof_ipc_comp comp;
+	struct sof_ipc_comp_config config;
+	/* selector supports 1 input and 1 output */
+	uint32_t input_channels_count; /**< accepted values 2 or 4 */
+	uint32_t output_channels_count; /**< accepted values 1 or 2 or 4 */
+	/* note: if 2 or 4 output channels selected the component works in a bypass 
+	 * mode*/
+	uint32_t selected_channel; /**< 0..3 */
```

Comment:
@lbetlej what does bypass mode mean? is it the same as passthrough mode? 
Then is the selected_channel valid only if the output_channels_count is 1?

Reply:
2 x yes , I've updated the comment